### PR TITLE
Fixed issue #18415: Copy Survey: New Title shouldn't be mandatory

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/SurveyAdministrationController.php [hunk 1]
```diff
@@ -2128,7 +2125,7 @@ public function actionCopy()
                 if (!$iSurveyID) {
                     $aData['sErrorMessage'] = gT("No survey ID has been provided. Cannot copy survey");
                     $aData['bFailed'] = true;
-                } elseif (!Survey::model()->findByPk($iSurveyID)) {
+                } elseif (!$sourceSurvey = Survey::model()->findByPk($iSurveyID)) {
```
Comment (Shnoulle) [inline application/controllers/SurveyAdministrationController.php:2128]: Personally i'm for separate the set and the test.

**Review by Shnoulle — CHANGES_REQUESTED** [review]
I'm not for the test and set at same time here. Since there vare already a lot of if/elseif/elseif/else add a complication.

**Comment by gabrieljenik** [comment]
> I'm not for the test and set at same time here. Since there vare already a lot of if/elseif/elseif/else add a complication.

Yes, I know.
Not the best. But if not, I would had to do some code reorganization, and wanting to avoid it just for this small story.

File: application/controllers/SurveyAdministrationController.php [hunk 2]
```diff
@@ -2140,6 +2137,11 @@ public function actionCopy()
                 } else {
                     Yii::app()->loadHelper('export');
                     $copysurveydata = surveyGetXMLData($iSurveyID, $aExcludes);
+                    if (empty(Yii::app()->request->getPost('copysurveyname'))) {
+                        $sNewSurveyName = $sourceSurvey->currentLanguageSettings->surveyls_title;
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:2141]: > $sourceSurvey = Survey::model()->findByPk($iSurveyID);
> $sNewSurveyName = $sourceSurvey->currentLanguageSettings->surveyls_title;

I didn't want to query the DB again. $sourceSurvey was already fetched couple of lines above when validating the survey id.

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:2141]: You are totally right, except for Survey :)

`Survey::model()->findByPk($iSurveyID)` use a static var 

https://github.com/LimeSurvey/LimeSurvey/blob/3215a764bf4033a490d70160b3f4eb2b3951b478/application/models/Survey.php#L981

It's not an issue (for Survey)

**Comment by Shnoulle** [comment]
> > I'm not for the test and set at same time here. Since there vare already a lot of if/elseif/elseif/else add a complication.
> 
> Yes, I know. Not the best. But if not, I would had to do some code reorganization, and wanting to avoid it just for this small story.

Since select are required

```
$iSurveyID = sanitize_int(Yii::app()->request->getParam('copysurveylist'));
if (!$iSurveyID) {
    thrown 400 (bad request)
}
$sourceSurvey = Survey::model()->findByPk($iSurveyID);
if(empty($sourceSurvey)) {
//etc …
```

**Comment by gabrieljenik** [comment]
Cosidering that the survey record is kind of  cached inside the model, I think that we can do a better fix without needing to do a whole code reorg.

**Comment by Shnoulle** [comment]
> Cosidering that the survey record is kind of cached inside the model, I think that we can do a better fix without needing to do a whole code reorg.

See this another comment too https://github.com/LimeSurvey/LimeSurvey/pull/2709#issuecomment-1316514411

dropdown is mandatory (`required="required"`) then to get here witth an empty sid (or a string for sid) : user must hack HTML (or have a Man in the middle attack) : throw a 400 request in place of if/elseif :)

**Comment by gabrieljenik** [comment]
> throw a 400 request in place of if/elseif :)

That's just one of the blocks in the chain of if/elseif.
I want to avoid code reorganization. If not bigger risk of unexpected errors.
Risk vs Benefit balance doesn't seem to good for me here

**Comment by Shnoulle** [comment]
It's for $sourceSurvey = Survey::model()->findByPk($iSurveyID); :)

PS : since we already set to int : we can call `Survey::model()->findByPk(0);`  : it's OK in all DB
PS2:  `Survey::model()->findByPk('STRING');` break with an error 500 for PGsql

File: application/controllers/SurveyAdministrationController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:2142]: Check now :)

File: application/views/surveyAdministration/tabCopy_view.php [hunk 3]
```diff
@@ -27,11 +27,11 @@
                     <div class="form-group">
                         <label for='copysurveyname' class=" control-label"><?php echo  eT("New survey title:"); ?> </label>
                         <div class="">
-                            <input type='text' id='copysurveyname' size='82' maxlength='200' name='copysurveyname' value='' required="required" class="form-control" />
+                            <input type='text' id='copysurveyname' size='82' maxlength='200' name='copysurveyname' value='' class="form-control" />
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/views/surveyAdministration/tabCopy_view.php]: > > Best : $sourceSurvey->currentLanguageSettings->surveyls_title
> 
> The source survey is selected from a dropdown on this screen.
> 

Arg, i'm a silly boy ;)

File: application/views/surveyAdministration/tabCopy_view.php [hunk 4]
```diff
@@ -27,11 +27,11 @@
                     <div class="form-group">
                         <label for='copysurveyname' class=" control-label"><?php echo  eT("New survey title:"); ?> </label>
                         <div class="">
-                            <input type='text' id='copysurveyname' size='82' maxlength='200' name='copysurveyname' value='' required="required" class="form-control" />
+                            <input type='text' id='copysurveyname' size='82' maxlength='200' name='copysurveyname' placeholder="Copy original survey title" value='' class="form-control" />
```
Comment (Shnoulle) [inline application/views/surveyAdministration/tabCopy_view.php:30]: `eT("Copy original survey title")`

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Just translation :)
Thank you.

File: application/views/surveyAdministration/tabCopy_view.php (hunk 4 above)
Comment (gabrieljenik) [reply application/views/surveyAdministration/tabCopy_view.php]: Updated!

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**